scripts/singleWindow.js

```diff
@@ -1,191 +1,38 @@
-                    (rl_title.length > 18 ? rl_title.substring(0, 15) + '...' : rl_title) +
-                    '</a></li>';
-                }
-                html += '</ul>';
-            }
-            document.getElementById("show_alexa_data").innerHTML = html;
-        }
-    };
-    http.send(null);
-}
-
-function get_whois(url){
-    var url=getUrlByParameter('url');
-    var host_url = url.replace(/^https{0,1}:\/\//, '').replace(/^www\./, '').replace(/\/.*/, '');
-    var whois_url="https://www.whoisxmlapi.com/whoisserver/WhoisService?domainName="+host_url+"&username=anishkumarsarangi&password=archiveit";
-    var http = new XMLHttpRequest();
-    http.open("GET", whois_url, true);
-    http.onreadystatechange = function() {
-        if (this.readyState == 4 && this.status == 200) {
-            var html = "<b>"+"<span class='color_code_whois'>"+ host_url +'</span>'+"</b><br/><b>Domain-Name: </b>";
-            var xmldata = http.responseXML.documentElement;
-            if (xmldata.getElementsByTagName("domainName")){
-                html +="<span class='color_code_whois'>"+xmldata.getElementsByTagName("domainName")[0].innerHTML+"</span>";
-            } 
-            else {
-                html += "N/A";
-            }
-            if(xmldata.getElementsByTagName("registrarName")[0]){
-                html += '<br/>'+'<b>Registrar: </b>' +"<span class='color_code_whois'>"+
-                    xmldata.getElementsByTagName('registrarName')[0].innerHTML;
-            }
-            if(xmldata.getElementsByTagName("rawText")){
-                html += '<br/><br/>'+"<span style='color:black'>"+
-                    xmldata.getElementsByTagName('rawText')[0].innerHTML;
+    var url = getUrlByParameter('url');
+    url = url.replace(/^https?:\/\//, '');
+    $.get(alexa_url + url, function (xml) {
+        var name = xml.getElementsByTagName("ALEXA")[0].getAttribute('URL');
+        $("#alexa_name").text(name);
+        if (xml.getElementsByTagName("POPULARITY")) {
+            var rank = xml.getElementsByTagName("POPULARITY")[0].getAttribute('TEXT');
+            $("#alexa_rank").text(rank);
+        }
+        if (xml.getElementsByTagName("COUNTRY")[0]) {
+            var country = xml.getElementsByTagName('COUNTRY')[0].getAttribute('NAME');
+            $("#alexa_country").text(country);
+        }
+        var rl = xml.getElementsByTagName('RL');
+        if (rl.length > 0) {
+            for (var i = 0, len = rl.length; i < len && i < 5; i++) {
+                var title = rl[i].getAttribute('TITLE');
+                var href = rl[i].getAttribute('HREF');
+                var list_item = $("<li>");
+                var link = $("<a>").attr('href', "http://" + href).attr('target', '_blank').attr('class', 'rl-a').text(title.length > 18 ? title.substring(0, 15) + '...' : title);
+                list_item.append(link);
+                $("#alexa_list").append(list_item);
```

Comment:
This can be written much more efficiently:
```
$('#alexa_list').append(
  $('<li>').append(
    $("<a>").attr('href', "http://" + href).attr('target', '_blank').attr('class', 'rl-a').text(title.length > 18 ? title.substring(0, 15) + '...' : title)
  )
)
```

Reply:
Done !!